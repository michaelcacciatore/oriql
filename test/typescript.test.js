const compileTypescript = require('../compile/typescript');

const schema = require('./schema/schema');

describe('Typescript', () => {
  it('should compile correctly', () => {
    const compiled = compileTypescript([schema]);

    expect(compiled).toMatchSnapshot();
  });
});
