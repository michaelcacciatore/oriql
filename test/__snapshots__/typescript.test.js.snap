// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typescript should compile correctly 1`] = `
"export namespace PropTypes {
  export interface testQuery {
    foo?: string | null | undefined;
    lol?: Array<string> | null | undefined;
    bar?: string;
    test:
      | Array<{
          michael?: string;
          brielle?: string | null | undefined;
          robert:
            | {
                itworks?: string | null | undefined;
                theresa?: string | null | undefined;
              }
            | null
            | undefined;
        }>
      | null
      | undefined;
    mom:
      | {
          michael?: string;
          brielle?: string | null | undefined;
          robert:
            | {
                itworks?: string | null | undefined;
                theresa?: string | null | undefined;
              }
            | null
            | undefined;
        }
      | null
      | undefined;
    nestedNoName:
      | {
          hasObject?: string | null | undefined;
          double: { isDouble?: string | null | undefined } | null | undefined;
        }
      | null
      | undefined;
    firstName?: string | null | undefined;
    employees:
      | Array<{
          id?: string | null | undefined;
          employee_name?: string | null | undefined;
          employee_salary?: string | null | undefined;
          employee_age?: string | null | undefined;
          profile_image?: string | null | undefined;
        }>
      | null
      | undefined;
  }
}
"
`;
